BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -1822,10 +1845,23 @@ void EndBoost()
                     if (gEmitter.Current == null) continue;
                     gEmitter.Current.emit = false;
                 }
+            
+            if (BDArmorySettings.MISSILE_FUEL_DEDUCTION)
+            {
+                float deduction = (boosterMass < 0) ? 0 : boosterMass;
+                if (BDArmorySettings.DEBUG_MISSILES) Debug.Log($"[BDArmory.Missile.FuelDeduction]: Error total mass wasn't deducted the final mass should be {initialMass - boosterMass} but is {part.mass}");
```

Comment:
This log statement should be inside the `if` block below. However, this check should not be necessary.
If the `boosterMass` should be fully used up at this point (i.e., there's no situation where only part of the boost mass is used), then you can just set it to the correct value (as mentioned in DMs on discord). If not, then it shouldn't be setting it.

Also, when comparing `float`s, you should not use equality/inequality (unless it's being compared with something like `float.MaxValue`), but rather check that the absolute difference is/isn't within some tolerance.